### PR TITLE
Nicer cli hashes

### PR DESCRIPTION
**Pull request by peterebden**
Rather than passing as `--hash` and `--size`, parse them like URL fragments from the browser (which is what I often have if I'm pasting in, and is intuitive enough otherwise).

For example:
```
plz run //discern -- show -s localhost:7775 79d6c0218ddf8ab4869a69b5cc613ceb86d89ffc419637301e53b8ff950489f5/146
```
vs before.
```
plz run //discern -- show -s localhost:7775 --hash 79d6c0218ddf8ab4869a69b5cc613ceb86d89ffc419637301e53b8ff950489f5 --size 146
```

### CODE REVIEW

**Comment by Tatskaari** [comment]
Ah I thought this before. It's annoying when copying around.

File: cli/cli.go [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+	"regexp"
+	"strconv"
+
+	pb "github.com/bazelbuild/remote-apis/build/bazel/remote/execution/v2"
+	"github.com/peterebden/go-cli-init"
+)
+
+var log = cli.MustGetLogger()
+
+var actionRe = regexp.MustCompile("([0-9a-fA-F]+)/([0-9]+)/?")
```
Comment (Tatskaari) [inline cli/cli.go:15]: nit: Don't really need a regex here. We could just strings.split on /

File: cli/cli.go [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+// An Action represents a combined hash / size pair written like
+// ff17a4efe382e245491d6a9f1ac6bf3adce454f7e4a5559a3579c3856edf1381/122
+// This is a bit more concise than passing them with flags.
+type Action struct {
+	Hash string
+	Size int
+}
+
+func (a *Action) UnmarshalFlag(in string) error {
+	matches := actionRe.FindStringSubmatch(in)
```
Comment (Tatskaari) [inline cli/cli.go:27]: if strings.contains(in, "/") would do it

File: cli/cli.go (hunk 1 above)
Comment (peterebden) [reply cli/cli.go:15]: Yeah but the regex also handles the restricted character sets; I don't think there's a trivial way of doing that.

I'd do something else if performance was an issue but it's only CLI parsing so it's not like we're gonna need to consume thousands of these.

File: cli/cli.go (hunk 2 above)
Comment (peterebden) [reply cli/cli.go:27]: yeah but then I'd have to do this anyway to get the submatches

**Pull request merged**